internal/server/iigointernal/executive.go

```diff
@@ -47,23 +47,31 @@ func (e *executive) setAllocationRequest(resourceRequests map[shared.ClientID]sh
 	e.ResourceRequests = resourceRequests
 }
 
+// setGameState is used for setting the game state of the executive branch
+// Called for testing.
+func (e *executive) setGameState(g *gamestate.GameState) {
+	e.gameState = g
+}
+
 // Get rules to be voted on to Speaker
 // Called by orchestration at the end of the turn
 func (e *executive) getRuleForSpeaker() shared.PresidentReturnContent {
-	e.budget -= serviceCharge
-	return e.clientPresident.PickRuleToVote(e.RulesProposals)
-}
+	if !e.incurServiceCharge(actionCost.GetRuleForSpeakerActionCost) {
+		return shared.PresidentReturnContent{
+			ContentType:  shared.PresidentRuleProposal,
+			ProposedRule: "",
+			ActionTaken:  false,
+		}
+	}
```

Comment:
getRuleForSpeaker() does not return a "Insufficient budget" error but broadcastTaxation does? Should this not also return an error if the cost cannot be deducted?